aws/data_source_aws_ec2_instance_spot_price.go

```diff
@@ -0,0 +1,106 @@
+package aws
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceAwsEc2InstanceSpotPrice() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsEc2InstanceSpotPriceRead,
+
+		Schema: map[string]*schema.Schema{
+			"filter": dataSourceFiltersSchema(),
+			"instance_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"spot_price": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"spot_price_timestamp": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsEc2InstanceSpotPriceRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	now := time.Now()
+	input := &ec2.DescribeSpotPriceHistoryInput{
+		StartTime: &now,
+	}
+
+	if v, ok := d.GetOk("instance_type"); ok {
+		instanceType := v.(string)
+		input.InstanceTypes = []*string{
+			aws.String(instanceType),
+		}
+	}
+
+	if v, ok := d.GetOk("availability_zone"); ok {
+		availabilityZone := v.(string)
+		input.AvailabilityZone = aws.String(availabilityZone)
+	}
+
+	if v, ok := d.GetOk("filter"); ok {
+		input.Filters = buildAwsDataSourceFilters(v.(*schema.Set))
+	}
+
+	var foundSpotPrice []*ec2.SpotPrice
+
+	for {
+		output, err := conn.DescribeSpotPriceHistory(input)
+
+		if err != nil {
+			return fmt.Errorf("error reading EC2 Spot Price History: %w", err)
+		}
+
+		if output == nil {
+			break
+		}
+
+		for _, instanceSpotPrice := range output.SpotPriceHistory {
+			if instanceSpotPrice == nil {
+				continue
+			}
+
+			foundSpotPrice = append(foundSpotPrice, instanceSpotPrice)
+		}
```

Comment:
This can be simplified to the following 😄 

```suggestion
		foundSpotPrice = append(foundSpotPrice, output.SpotPriceHistory...)
```

Reply:
simplified the usage of this greatly in c0993d124ea04d7f3ccf27cd5bd275dbf7ac6a05 :+1: